src/renderer/components/+cluster-settings/components/cluster-icon-setting.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { Cluster } from "../../../../main/cluster";
+import { ClusterStore, clusterStore } from "../../../../common/cluster-store"
+import { Icon } from "../../icon";
+import { FilePicker } from "../../../components/file";
+import { autobind } from "../../../utils";
+import { Button } from "../../button";
+
+enum ValueStatus {
+  CLEAN = "clean",
+  ERROR = "error",
+}
+
+interface Props {
+    cluster: Cluster;
+}
+
+interface State {
+    status: ValueStatus;
+    errorText?: string;
+}
+
+export class ClusterIconSetting extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      status: ValueStatus.CLEAN,
+    }
+  }
+
+  @autobind()
+  async onIconPick([file]: File[]) {
+    const { cluster } = this.props;
+    
+    try {
+      cluster.preferences.icon = "";
+      if (file) {
+        console.log(`using ${file.name}`);
+
+      }
+      const buf = Buffer.from(await file.arrayBuffer());
+      cluster.preferences.icon = `data:image/jpeg;base64, ${buf.toString('base64')}`;
```

Comment:
I tried that, but it doesn't work on development due to security concerns. I found that this works even better since we don't have to do file cleanup and we don't have to disable webSecurity during development.